File: TFG/ftp_tfg/lib/domain/repositories/monitoring_repository.dart
import '../entities/system_alert.dart';
import '../entities/file_version.dart';
import '../entities/system_event.dart';
import '../entities/system_health_summary.dart';
import '../entities/sync_record.dart';

abstract class MonitoringRepository {
  Future<void> recordEvent(SystemEvent event);
  Future<int> createAlert(SystemAlert alert);
  Future<int> recordFileVersion(FileVersion version);
  Future<List<SystemEvent>> getRecentEvents(String ownerId, {int limit = 20});
  Future<List<SystemAlert>> getActiveAlerts(String ownerId, {int limit = 10});
  Future<List<SyncRecord>> getRecentSyncs(String ownerId, {int limit = 20});
  Future<List<FileVersion>> getRecentFileVersions(
    String ownerId, {
    int limit = 12,
  });
  Future<FileVersion?> getLatestFileVersion(
    String ownerId,
    int profileId,
    String filePath,
  );
  Future<List<FileVersion>> getFileVersionHistory(
    String ownerId,
    int profileId,
    String filePath, {
    int limit = 20,
  });
  Future<void> acknowledgeAlert(int alertId, String ownerId);
  Future<SystemHealthSummary> getHealthSummary(String ownerId);
}

File: TFG/ftp_tfg/hotftp_api/src/domain/repositories/monitoring-repository.ts
import type { FileVersion } from '../entities/file-version.js';
import type { HealthSummary } from '../entities/health-summary.js';
import type { SyncRecord } from '../entities/sync-record.js';
import type { SystemAlert } from '../entities/system-alert.js';
import type { SystemEvent } from '../entities/system-event.js';

export interface MonitoringRepository {
  recordEvent(event: SystemEvent): Promise<void>;
  createAlert(alert: SystemAlert): Promise<number>;
  recordFileVersion(version: FileVersion): Promise<number>;
  getRecentEvents(ownerId: string, limit?: number): Promise<SystemEvent[]>;
  getActiveAlerts(ownerId: string, limit?: number): Promise<SystemAlert[]>;
  recordSync(record: SyncRecord): Promise<void>;
  getHealthSummary(ownerId: string): Promise<HealthSummary>;
  getSyncHistory(ownerId: string): Promise<SyncRecord[]>;
  getRecentFileVersions(ownerId: string, limit?: number): Promise<FileVersion[]>;
  getLatestFileVersion(
    ownerId: string,
    profileId: number,
    filePath: string,
  ): Promise<FileVersion | null>;
  getFileVersionHistory(
    ownerId: string,
    profileId: number,
    filePath: string,
    limit?: number,
  ): Promise<FileVersion[]>;
  acknowledgeAlert(alertId: number, ownerId: string): Promise<void>;
}

